backend/src/test/java/ca/bc/gov/hlth/mohums/integration/MoHUmsIntegrationTests.java

```diff
@@ -91,12 +94,107 @@ public void clientsAuthorized() throws Exception {
+                        .build())
+                .header("Authorization", "Bearer " + jwt)
+                .exchange()
+                .expectStatus().isOk()
+                .expectBodyList(Object.class)
+                .returnResult()
+                .getResponseBody();
+
+        Assertions.assertThat(filteredUsers).isNotEmpty();
+        Assertions.assertThat(allUsers).containsAll(filteredUsers);
+    }
+
+    @Test
+    public void searchByEmailAndOrganization() throws Exception {
+        // Given a test user with Org. ID present...
+        webTestClient
+                .post()
+                .uri("/users")
+                .contentType(MediaType.APPLICATION_JSON)
+                .bodyValue("{\"enabled\":true,\"username\":\"testWithOrgId\",\"firstName\":\"Test\",\"lastName\":\"WithOrgId\",\"email\":\"test@domain.com\",\"emailVerified\":\"\",\"attributes\":{\"org_details\":[\"{\\\"id\\\":\\\"00001763\\\"}\"]}}")
+                .header("Authorization", "Bearer " + jwt)
+                .exchange()
+                .expectStatus().value(status -> Assertions.assertThat(status).isIn(
+                        // CREATED is returned when the user does not already exist.
+                        HttpStatus.CREATED.value(),
+                        // CONFLICT is returned when the user already exists.
+                        HttpStatus.CONFLICT.value()));
+
+        // ... and another test user with the same e-mail but no Org. ID, ...
+        webTestClient
+                .post()
+                .uri("/users")
+                .contentType(MediaType.APPLICATION_JSON)
+                .bodyValue("{\"enabled\":true,\"username\":\"testWithoutOrgId\",\"firstName\":\"Test\",\"lastName\":\"WithoutOrgId\",\"email\":\"test@domain.com\",\"emailVerified\":\"\",\"attributes\":{}}")
+                .header("Authorization", "Bearer " + jwt)
+                .exchange()
+                .expectStatus().value(status -> Assertions.assertThat(status).isIn(
+                        // CREATED is returned when the user does not already exist.
+                        HttpStatus.CREATED.value(),
+                        // CONFLICT is returned when the user already exists.
+                        HttpStatus.CONFLICT.value()));
+
+        // ... a search by that e-mail address and filtering by Org. ID...
+        final List<Object> filteredUsers = webTestClient
+                .get()
+                .uri(uriBuilder -> uriBuilder
+                        .path("/users")
+                        .queryParam("email", "test@domain.com")
+                        .queryParam("org", "00001763")
+                        .build())
+                .header("Authorization", "Bearer " + jwt)
+                .exchange()
+                .expectStatus().isOk()
+                .expectBodyList(Object.class)
+                .returnResult()
+                .getResponseBody();
+
+        // ... should return at least one filtered user
```

Comment:
Nice test and good use of commenting. I try not to "over-comment", but these really add clarity to your test case.